### PR TITLE
Feature/WF 458 e heading

### PR DESCRIPTION
**Pull request by boehlerlukas**
This pull request adds a e-heading component. I created a example page so you can test in all browsers.

Please note: because of some reset styles of vue-styleguidist, the styles are not 100% valid in the component styleguide (e.g. buttons have a black font on iOS Safari). The demo page shows the correct styles.

### Ticket

https://winterhalter-fenner.atlassian.net/browse/WF-458

### Browser testing

http://localhost:6060/styleguide#!/e-heading
http://localhost:8080/styleguide/headings

### Checklist

- [ ]  Tested all links in project relevant browsers
- [ ]  Tested all links in different screen sizes
- [ ]  Did run automated tests and linters
- [ ]  Did assign ticket
- [ ]  Double checked target branch
- [ ] Review/Test checklist

- [ ]  Did run automated tests and linters
- [ ]  Did review code
- [ ]  Tested all links in project relevant browsers
- [ ]  Tested all links in different screen sizes
- [ ]  Re-assign ticket to developer

### CODE REVIEW

File: app/components/e-heading.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+### Alternate color
+
+```
+<div>
+  <e-heading level="h1" color="darkgrey">Headline-1 Regular (components Title)</e-heading>
+  <e-heading level="h2" color="green">Headline-2 (Information Title)</e-heading>
+  <e-heading level="h3" color="darkgrey">Headline-3 (Information Title)</e-heading>
+  <e-heading level="h4" color="green">Headline-4 (Navigation Title aktive)</e-heading>
```
Comment (geroldboehler) [inline app/components/e-heading.md]: Use english text

File: app/components/e-heading.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+### With underline
+
+```
+<div>
+  <e-heading level="h1" underline="true">Headline-1 Regular (components Title)</e-heading>
+  <e-heading level="h2" underline="true">Headline-2 (Information Title)</e-heading>
+  <e-heading level="h3" underline="true">Headline-3 (Information Title)</e-heading>
+  <e-heading level="h4" underline="true">Headline-4 (Navigation Title aktive)</e-heading>
```
Comment (geroldboehler) [inline app/components/e-heading.md]: Use english text

File: app/components/e-heading.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+### Alternate font weight
+
+```
+<div>
+  <e-heading level="h1" weight="bold">Headline-1 Regular (components Title)</e-heading>
+  <e-heading level="h2" weight="semibold">Headline-2 (Information Title)</e-heading>
+  <e-heading level="h3" weight="bold">Headline-3 (Information Title)</e-heading>
+  <e-heading level="h4" weight="normal">Headline-4 (Navigation Title aktive)</e-heading>
```
Comment (geroldboehler) [inline app/components/e-heading.md]: Use english text

File: app/components/e-heading.vue [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        },
+      },
+
+      /**
+       * Defines the level of the heading (h1-h6)
+       */
+      level: {
+        type: String,
+        required: true,
+      },
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Validate input

File: app/components/e-heading.vue [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+      /**
+       * Defines if the heading should have an underline
+       */
+      underline: {
+        type: Boolean,
+        default: null,
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Set default to true

File: app/components/e-heading.vue [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+      },
+
+      /**
+       * Defines the font weight
+       */
+      weight: {
+        type: String,
+        default: null,
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Set normal as default

File: app/components/e-heading.vue [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+     * @param   {function}    createElement   Vue helper
+     *
+     * @returns {*}
+     */
+    render(createElement) {
+      const element = this.$props.level;
+      const attributes = {
+        class: this.b({
+          color: this.$props.color,
+          underline: this.$props.underline === 'true',
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please remove "=== 'true'" since underline is boolean already

File: app/components/e-heading.vue [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+          weight: this.$props.weight,
+        }),
+        attrs: {
+
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: add ...this.$attrs,

File: app/components/e-heading.vue [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+        attributes,
+        this.$slots.default,
+      );
+    },
+  };
+</script>
+
+<style lang="scss">
+  .e-heading {
+    font-family: "Source Sans Pro", sans-serif;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+    font-weight: normal;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+    color: $color-secondary--1;
+    letter-spacing: 2.03px;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Where is this coming from?

File: app/components/e-heading.vue [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+    line-height: 23px;
+
+    &.e-heading--weight-normal {
+      font-weight: normal;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    &.e-heading--weight-semibold {
+      font-weight: 600;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    &.e-heading--weight-bold {
+      font-weight: bold;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+      color: $color-grayscale--200;
+    }
+
+    &.e-heading--color-green {
+      color: $color-secondary--3;
+    }
+  }
+
+  h1.e-heading {
+    font-size: 18px;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  h2.e-heading {
+    font-size: 16px;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  h3.e-heading {
+    font-size: 14px;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  h4.e-heading {
+    font-size: 14px;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+    font-weight: bold;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  h5.e-heading {
+    font-size: 14px;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+    font-weight: 600;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 22]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  h6.e-heading {
+    font-size: 14px;
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please use sass variables

File: app/components/e-heading.vue [hunk 23]
```diff
@@ -0,0 +1,164 @@
...
+
+<style lang="scss">
+  .e-heading {
+    font-family: "Source Sans Pro", sans-serif;
+    font-weight: normal;
+    color: $color-secondary--1;
+    letter-spacing: 2.03px;
+    line-height: 23px;
+
+    &.e-heading--weight-normal {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please define separate selector &--weight-normal

File: app/components/e-heading.vue [hunk 24]
```diff
@@ -0,0 +1,164 @@
...
+      font-weight: normal;
+    }
+
+    &.e-heading--weight-semibold {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please define separate selector &--weight-semibold

File: app/components/e-heading.vue [hunk 25]
```diff
@@ -0,0 +1,164 @@
...
+      font-weight: 600;
+    }
+
+    &.e-heading--weight-bold {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please define separate selector &--weight-bold

File: app/components/e-heading.vue [hunk 26]
```diff
@@ -0,0 +1,164 @@
...
+      font-weight: bold;
+    }
+
+    &.e-heading--underline {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please define separate selector &--underline

File: app/components/e-heading.vue [hunk 27]
```diff
@@ -0,0 +1,164 @@
...
+      text-decoration: underline;
+    }
+
+    &.e-heading--color-blue {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please define separate selector &--color-blue

File: app/components/e-heading.vue [hunk 28]
```diff
@@ -0,0 +1,164 @@
...
+      color: $color-secondary--1;
+    }
+
+    &.e-heading--color-darkgrey {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please define separate selector &--color-darkgray

File: app/components/e-heading.vue [hunk 29]
```diff
@@ -0,0 +1,164 @@
...
+      color: $color-grayscale--200;
+    }
+
+    &.e-heading--color-green {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Please define separate selector &--color-green

File: app/components/e-heading.vue [hunk 30]
```diff
@@ -0,0 +1,164 @@
...
+      color: $color-secondary--3;
+    }
+  }
+
+  h1.e-heading {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Don't use tag names as selectors, please set a class name for H and move it up inside .e-heading

File: app/components/e-heading.vue [hunk 31]
```diff
@@ -0,0 +1,164 @@
...
+    font-size: 18px;
+  }
+
+  h2.e-heading {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Don't use tag names as selectors, please set a class name for H and move it up inside .e-heading

File: app/components/e-heading.vue [hunk 32]
```diff
@@ -0,0 +1,164 @@
...
+    font-size: 16px;
+  }
+
+  h3.e-heading {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Don't use tag names as selectors, please set a class name for H and move it up inside .e-heading

File: app/components/e-heading.vue [hunk 33]
```diff
@@ -0,0 +1,164 @@
...
+    font-size: 14px;
+  }
+
+  h4.e-heading {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Don't use tag names as selectors, please set a class name for H and move it up inside .e-heading

File: app/components/e-heading.vue [hunk 34]
```diff
@@ -0,0 +1,164 @@
...
+    font-size: 14px;
+    font-weight: bold;
+  }
+
+  h5.e-heading {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Don't use tag names as selectors, please set a class name for H and move it up inside .e-heading

File: app/components/e-heading.vue [hunk 35]
```diff
@@ -0,0 +1,164 @@
...
+    font-size: 14px;
+    font-weight: 600;
+  }
+
+  h6.e-heading {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: Don't use tag names as selectors, please set a class name for H and move it up inside .e-heading

File: app/components/e-heading.vue [hunk 36]
```diff
@@ -0,0 +1,164 @@
...
+            'green',
+            'darkgrey',
+          ].includes(value);
+        },
+      },
+
+      /**
+       * Defines the level of the heading (h1-h6)
+       */
+      level: {
```
Comment (geroldboehler) [inline app/components/e-heading.vue]: If it is called "level", please use Number as type and just pass the numbers 1-6

File: app/components/e-heading.md [hunk 37]
```diff
@@ -0,0 +1,51 @@
+### Different Tags
```
Comment (geroldboehler) [inline app/components/e-heading.md]: Please just use "Heading X Lorem ipsum..." or similar as demo text.

Comment (geroldboehler) [reply app/components/e-heading.md]: Please use Heading, not Headline

File: app/components/e-heading.md (hunk 1 above)
Comment (boehlerlukas) [reply app/components/e-heading.md]: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua auch ok?

File: app/components/e-heading.vue (hunk 11 above)
Comment (boehlerlukas) [reply app/components/e-heading.vue]: Das kommt von: https://projects.invisionapp.com/d/main#/console/13587539/292245532/inspect :)

File: app/components/e-heading.vue (hunk 36 above)
Comment (boehlerlukas) [reply app/components/e-heading.vue]: Re-named it to tagName :)

**Comment by boehlerlukas** [comment]
@geroldboehler the latest version ist pushed. You can also checkout the test/demo page at http://localhost:8080/styleguide/headings

**Comment by geroldboehler** [comment]
Please check off checklist items.

**Pull request closed without merging**